### DISCUSSION TITLE
Hidden Markov Model with Control

### DISCUSSION THREAD

**Discussion by kobus78** [ReactiveBayes/RxInfer.jl#308] [Q&A] [answered]
Please have a look at this post:
https://learnableloop.com/posts/HMMwithControl_PORT.html
This post starts with the 'Hidden Markov Model' RxInfer example and tries to add a control framework around it. I have inserted some of the structures towards this end but I'm not quite sure what the next steps are. If anyone at the BIASLab or Lazy Dynamics or anyone else would like to give me some pointers, I will be very grateful. I think having a control example making use of a categorical control space will be very helpful to new users of RxInfer. We are currently working on a number of projects at the Active Inference Institute that definitely needs this.

**Comment by albertpod** [comment]
Thanks @kobus78!

@raganapa will have a look into this one!

**Reply by kobus78** [reply]
Thanks a lot @albertpod. @raganapa, I have progressed a little bit further. Please have a look at part 2:
https://learnableloop.com/posts/HMMwithControl2_PORT.html
Please note that I got a breakage towards the end. Not sure how to address this.
Thanks for your help!

**Reply by raganapa** [reply]
@kobus78  I am currently on holidays, I will start looking into in from 10/Jun.

**Comment by kobus78** [comment]
OK, thanks Raaja.

**Comment by albertpod** [comment, marked as answer]
Hi @kobus78 @LearnableLoopAI, I guess it's done:
https://github.com/ReactiveBayes/ReactiveMP.jl/pull/439
